### PR TITLE
Feature/df 1544 competence elements deletion

### PR DESCRIPTION
**Pull request by emina-str**
translations will come in couple of days ...

### CODE REVIEW

File: alv-portal-ui/translations.csv [hunk 1]
```diff
@@ -1172,8 +1172,11 @@ portal.competence-catalog.competence-elements.overview.search.placeholder,"Suche
...
+portal.competence-catalog.competence-elements.delete-modal.yes,"Ja, löschen","Yes, delete","Oui, supprimer","Sì, cancellare",
+portal.competence-catalog.competence-elements.delete-modal.title,Kompetenz-Bestandteil löschen?,Delete skill component?,Effacer un composant de compétence?,Cancellare elemento della competenza?,
+portal.competence-catalog.competence-elements.delete-modal.confirmation,"Möchten Sie dieses Kompetenz-Bestandteil ""{{ competenceElementId }}"" wirklich löschen? Wenn Sie auf ""Ja, löschen"" klicken, wird diese Kompetenz-Bestandteil unwiderruflich gelöscht.","Do you really want to delete the skill component ""{{ competenceElementId }}""? If you click ""Yes, delete"", this skill component will be permanently deleted.","Souhaitez-vous vraiment supprimer ce composant de compétence ""{{ competenceElementId }}""? Si vous cliquez sur ""Oui, supprimer"", ce composant de compétence sera définitivement supprimé.","Vuoi davvero cancellare questo elemento della competenza ""{{ competenceElementId }}""? Cliccando su ""Sì, cancellare"" questo elemento sarà definitivamente soppresso insieme con le sue attribuzioni.",
```
Comment (cjpsycos) [inline alv-portal-ui/translations.csv]: missing <br>

File: alv-portal-ui/src/app/competence-catalog/shared/deletion/competence-element-delete/competence-element-delete.component.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<ng-container *ngIf="{competenceSets: competenceSets$ | async} as observables">
```
Comment (birom) [inline alv-portal-ui/src/app/competence-catalog/shared/deletion/competence-element-delete/competence-element-delete.component.html]: I reopened the https://alv-ch.atlassian.net/browse/DF-927 technical debt issue.

IMHO the following would be a cleaner solution:
```html
<ng-container *ngIf="competenceSets$ | async as competenceSets">
   <alv-modal (cancelAction)="cancel()"
             (primaryAction)="delete()"
             [showPrimaryButton]="!competenceSets.length"
             cancelLabel="entity.action.cancel"
             primaryLabel="portal.global.delete-confirm"
             modalTitle="portal.competence-catalog.competence-elements.delete-modal.title">
    <ng-container *ngIf="competenceSets.length; else noReferences">
      <alv-alert [hideDismiss]="true"
                 [notification]="warning">
      </alv-alert>
......
```

File: alv-portal-ui/src/app/competence-catalog/shared/deletion/competence-element-delete/competence-element-delete.component.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<ng-container *ngIf="(competenceSets$ | async) as competenceSets">
+  <alv-modal (cancelAction)="cancel()"
+             (primaryAction)="delete()"
+             [showPrimaryButton]="!competenceSets?.length"
```
Comment (birom) [inline alv-portal-ui/src/app/competence-catalog/shared/deletion/competence-element-delete/competence-element-delete.component.html]: ``competenceSets`` can not be null or undefined here. The Elvis operator is not needed:

```html
 <alv-modal (cancelAction)="cancel()"
             (primaryAction)="delete()"
             [showPrimaryButton]="competenceSets.length > 0"
```

File: alv-portal-ui/src/app/competence-catalog/shared/deletion/competence-element-delete/competence-element-delete.component.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+             cancelLabel="entity.action.cancel"
+             primaryLabel="portal.global.delete-confirm"
+             modalTitle="portal.competence-catalog.competence-elements.delete-modal.title">
+    <ng-container *ngIf="competenceSets?.length; else noReferences">
```
Comment (birom) [inline alv-portal-ui/src/app/competence-catalog/shared/deletion/competence-element-delete/competence-element-delete.component.html]: The Elvis operator is not needed.

**Pull request merged**